src/Razor/src/Microsoft.AspNetCore.Razor.Language/Intermediate/HtmlContentIntermediateNode.cs

```diff
@@ -9,6 +9,8 @@ public sealed class HtmlContentIntermediateNode : IntermediateNode
     {
         public override IntermediateNodeCollection Children { get; } = new IntermediateNodeCollection();
 
+        public bool IsEncoded { get; set; }
```

Comment:
Hmm, lets not put this on the `HtmlContentIntermediateNode` since it's public API and it's not absolute that the content is encoded i.e. `<p>John & Doe</p>`, and the info is only really needed for components. I'd decorate the nodes with annotations and pull them out in the following passes.

Reply:
You mean something like https://github.com/aspnet/AspNetCore-Tooling/pull/374/commits/8a75ea3e9bccd62a1560c3da6055811e6e9d6bd2#diff-fd3db52050596c91fd6f40795c08c4f8R16? This was in my first commit.  😄

Reply:
Yup, don't agree with @rynowak on that one for my reasons above. @rynowak did you feel strongly? This seems like a bad idea to me.

Only change I'd make is call the extension methods `HasEncodedContent` as well and mark the extension class as internal for now.

Reply:
I don't feel super strongly about what approach you take here. I gave the feedback I did because the original version seemed overly complex for something that *just* sets a property. 

How about you two discuss and come up with the best solution?

The only thing I ask is that you don't needlessly tostring the contents of the node like was in the previous iteration.

Reply:
Thanks guys. I'm going to bring back annotations, make it internal. Won't bring back the ToString part though. Sounds good @NTaylorMullen ?

Reply:
Sounds good to me!